src/symbolize/mod.rs

```diff
@@ -188,6 +229,30 @@ impl<'a> SymbolName<'a> {
     }
 }
 
+// With std enabled attempts to lossy convert bytes, on core uses
+// strict checking
+#[cfg(feature = "std")]
+fn format_symbol_name(fmt: fn(&str, &mut fmt::Formatter) -> fmt::Result,
+                      bytes: &[u8],
+                      f: &mut fmt::Formatter)
+    -> fmt::Result
+{
+    fmt(&*String::from_utf8_lossy(bytes), f)
+}
+
+#[cfg(not(feature = "std"))]
+fn format_symbol_name(fmt: fn(&str, &mut fmt::Formatter) -> fmt::Result,
+                      bytes: &[u8],
+                      f: &mut fmt::Formatter)
+    -> fmt::Result
+{
+    if let Ok(name) = str::from_utf8(bytes) {
```

Comment:
I think it's possible to use the `Err` here with [`error_len`](https://doc.rust-lang.org/std/str/struct.Utf8Error.html#method.error_len) to rig up something roughly equivalent to `from_utf8_lossy` to allow printing both and outside of std?